### PR TITLE
Minor docs and scripts improvements

### LINKED ISSUE #610: docs/user-guide update installation instructions
let's update the installation instructions to show install via docker, from gh releases and then from source (related #604)

### LINKED ISSUE #662: CI cargo doc build succeeds with a failed docs build
- in a PR: https://ci.heliax.dev/anoma/anoma/2003/4/3
- on master after merge: https://ci.heliax.dev/anoma/anoma/2005/3/3

### PR DESCRIPTION
**Pull request by SimSaladin**
- Fix a build error in the docs
- Silence ShellCheck warnings in shell scripts and make them exit on error

closes #610, closes #662

### CODE REVIEW

**Comment by Fraccaman** [comment]
pls update wasm

**Comment by Fraccaman** [comment]
pls update ci

File: docs/src/user-guide/intent-gossip-and-matchmaker.md [hunk 1]
```diff
@@ -108,4 +108,4 @@ fn add_intent(last_state: Vec<u8>, intent_id: Vec<u8>, intent_data: Vec<u8>) ->
 
 The matchmaker can keep some state between its runs. The state can be updated from within the matchmaker code with [`update_state` function](https://docs.anoma.network/rustdoc/anoma_vm_env/imports/matchmaker/fn.update_state.html) and received from the `last_state` argument.
 
-To find out about the interface available in a matchmaker and the library code used in the `mm_token_exch` implementation, please check out [Rust docs for `matchmaker_prelude`](/rustdoc/anoma_vm_env/matchmaker_prelude/index.html).
+To find out about the interface available in a matchmaker and the library code used in the `mm_token_exch` implementation, please check out [Rust docs for `matchmaker_prelude`](https://docs.anoma.network/rustdoc/anoma_vm_env/matchmaker_prelude/index.html).
```
Comment (tzemanovic) [inline docs/src/user-guide/intent-gossip-and-matchmaker.md:111]: The link has to include the branch, let's use master:
```suggestion
To find out about the interface available in a matchmaker and the library code used in the `mm_token_exch` implementation, please check out [Rust docs for `matchmaker_prelude`](https://docs.anoma.network/master/rustdoc/anoma_vm_env/matchmaker_prelude/index.html).
```

**Comment by Fraccaman** [comment]
pls update ci

**Comment by tzemanovic** [comment]
pls update ci

**Comment by tzemanovic** [comment]
pls update ci

File: scripts/install/get_tendermint.sh [hunk 2]
```diff
@@ -1,51 +1,47 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-#set -x
+set -euEo pipefail
```
Comment (tzemanovic) [inline scripts/install/get_tendermint.sh:3]: with ubuntu `GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)`, this only seems to work with:
```suggestion
set -Eo pipefail
```

**Comment by tzemanovic** [comment]
most commits have been merged via #661

**Review by tzemanovic — APPROVED** [review]

**Pull request merged**